fl4health/clients/moon_client.py

```diff
@@ -137,9 +161,9 @@ def compute_loss_and_additional_losses(
             "loss": loss,
         }
 
-        if self.contrastive_weight:
-            contrastive_loss = self.get_contrastive_loss(
-                features["features"], features["global_features"], features["old_features"]
+        if "old_features" in features and "global_features" in features:
```

Comment:
Maybe a quick comment to mention why we are unsqueezing 0th dimension of global features (ie only one positive pair expected)

Reply:
Should we also enforce that these keys are in the features dict? Is there any cases where we use MOON and not want to use the contrastive loss

Reply:
So these features will actually not be present for the server round. For example, there are no "old_features" in the first round, because there is no "previous round model" yet. I thought about using the server round instead of the presence of these features, but we don't currently pass the server round to this function. So thought I would do this for now and we can revisit this at a later stage.

Reply:
Ahhh yes, that is right. Makes sense to me.